src/v/redpanda/admin/server.cc

```diff
@@ -2106,6 +2108,31 @@ admin_server::update_user_handler(std::unique_ptr<ss::http::request> req) {
     co_return ss::json::json_return_type(ss::json::json_void());
 }
 
+ss::future<ss::json::json_return_type>
+admin_server::oidc_whoami_handler(std::unique_ptr<ss::http::request> req) {
+    auto auth_hdr = req->get_header("authorization");
+    if (!auth_hdr.starts_with(authz_bearer_prefix)) {
+        throw ss::httpd::base_exception{
+          "Invalid Authorization header",
+          ss::http::reply::status_type::unauthorized};
+    }
+
+    security::oidc::authenticator auth{_controller->get_oidc_service().local()};
+    auto res = auth.authenticate(auth_hdr.substr(authz_bearer_prefix.length()));
+
+    if (res.has_error()) {
+        throw ss::httpd::base_exception{
+          "Invalid Authorization header",
+          ss::http::reply::status_type::unauthorized};
+    }
```

Comment:
question: would it be any good to include the reason why the token was invalid in the response (I think `res` may contain such info)

Reply:
> question: would it be any good to include the reason why the token was invalid in the response (I think `res` may contain such info)

Since the endpoint is `whoami`, and the `authentication_header` has been taken from the request, it would already have performed a similar check and returned prior to getting here. Even if it did get here, I don't think it's wise to expose too much detail as to why a request failed, it gives an attacker information to reduce their search space.

If the request was more like `finger` then it would make sense for a superuser to be able to figure out what's wrong with the token, after they have passed their own auth. That was ruled out as a requirement.